operator/cmd/mesh/manifest-versions.go

```diff
@@ -79,6 +81,38 @@ func manifestVersions(args *rootArgs, mvArgs *manifestVersionsArgs, l *Logger) e
 	return nil
 }
 
+// CheckK8sVersion checks if the current client and server version is supported
+func CheckK8sVersion(opts *kubectlcmd.Options, l *Logger) error {
+	cK8sVer, sK8sVer, err := manifest.Version(opts)
+	if err != nil {
+		return fmt.Errorf("failed to read versions from client and cluster: %v", err)
+	}
+	compatibleMap, err := getVersionCompatibleMap("", binversion.OperatorBinaryGoVersion, l)
+	if err != nil {
+		return err
+	}
+	return checkK8sVersion(compatibleMap, cK8sVer, sK8sVer)
+}
+
+// checkK8sVersion checks if the client and server versions are supported in CompatibilityMapping
+func checkK8sVersion(vmap *version.CompatibilityMapping, cVer, sVer *goversion.Version) error {
+	if cVer == nil {
+		return fmt.Errorf("failed to get k8s client version, please check if kubectl is correctly installed")
+	}
+	if sVer == nil {
+		return fmt.Errorf("failed to get k8s server version, please check if the k8s context is configured")
+	}
+	if !vmap.K8sClientVersionRange.Check(cVer) {
+		return fmt.Errorf("failed to meet the minimal requirements of kubectl version for Istio %s",
+			binversion.OperatorCodeBaseVersion)
+	}
+	if !vmap.K8sServerVersionRange.Check(sVer) {
+		return fmt.Errorf("failed to meet the minimal requirements of k8s server version for Istio %s",
+			binversion.OperatorCodeBaseVersion)
+	}
+	return nil
+}
```

Comment:
imo this functionality would be better placed in some part of pkg (manifest?) rather than next to the manifest-versions command.

Reply:
yup, moved it to pkg/